gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineJdkRelease.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline.plugins;
+
+import com.google.common.collect.ImmutableList;
+import java.util.List;
+import org.gradle.api.JavaVersion;
+import org.gradle.api.Project;
+import org.gradle.api.tasks.compile.JavaCompile;
+
+/**
+ * <p>
+ * When using JDK 9+ to compile with a targetCompatibility less than JDK 9, this plugin adds compiler arguments
+ * per <a href="https://openjdk.java.net/jeps/247">JEP 247</a> to explicitly set the target JDK platform API
+ * to maintain binary compatibility.
+ * </p>
+ * <p>
+ * See also <a href="https://github.com/gradle/gradle/issues/2510">Gradle JDK release issue</a>.
+ * </p>
+ */
+public final class BaselineJdkRelease extends AbstractBaselinePlugin {
+
+    @Override
+    public void apply(Project project) {
+        this.project = project;
+        project.getPluginManager().withPlugin("java", plugin -> {
+            project.getTasks().withType(JavaCompile.class).configureEach(javaCompile -> {
+                try {
+                    configureJavaCompiler(javaCompile);
+                } catch (Throwable throwable) {
```

Comment:
What errors are we attempt to catch? Do we need the try-catch?

Reply:
I added this as I was debugging since this was my first gradle plugin so I'd have a place to set a debug breakpoint, and gradle was eating some exceptions, so wanted to make sure they were logged if one occurs